rgao77/tzs5dd_DS5111su24_lab_01
# Tokenizer

### Summary
This pull request adds three text processing functions (`clean_text`, `tokenize`, `count_words`) and sets up the project for testing and linting. 

### Changes Made
- Implemented `clean_text` function to remove punctuation and convert text to lowercase.
- Implemented `tokenize` function to split text into a list of words.
- Implemented `count_words` function to count the occurrences of each word in the text.
- Added unit tests for the functions in `test_text_processor.py`.
- Set up `Makefile` with tasks for setting up the virtual environment, running tests, and linting.
- Added `requirements.txt` for dependency management.

### Why These Changes Are Necessary
- These functions are foundational for processing text data for analysis.
- Unit tests ensure the functions work correctly.
- Setting up testing and linting helps maintain code quality and catch issues early.

### How to Test
1. Clone the repository and checkout the `tokenizer` branch.
2. Run `make setup` to set up the virtual environment and install dependencies.
3. Run `make test` to execute the unit tests.
4. Run `make lint` to check for coding standard violations.

### Additional Context
- The changes adhere to Python coding standards and best practices.
- All tests and lint checks pass with a perfect score.
